TESTS/netsocket/tls/tlssocket_endpoint_close.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2019, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "tls_tests.h""
+#include "mbed.h"
+#include "TLSSocket.h"
+#include "greentea-client/test_env.h"
+#include "unity/unity.h"
+#include "utest.h"
+
+using namespace utest::v1;
+
+namespace {
+static const int SIGNAL_SIGIO = 0x1;
+static const int SIGIO_TIMEOUT = 20000; //[ms]
+}
+
+static void _sigio_handler(osThreadId id)
+{
+    osSignalSet(id, SIGNAL_SIGIO);
+}
+
+static nsapi_error_t _tlssocket_connect_to_daytime_srv(TLSSocket &sock)
+{
+    SocketAddress tls_addr;
+
+    get_interface()->gethostbyname(MBED_CONF_APP_ECHO_SERVER_ADDR, &tls_addr);
+    tls_addr.set_port(2013);
+
+    nsapi_error_t err = sock.open(get_interface());
+    if (err != NSAPI_ERROR_OK) {
+        return err;
+    }
+
+    return sock.connect(tls_addr);
+}
+
+
+void TLSSOCKET_ENDPOINT_CLOSE()
+{
+    static const int MORE_THAN_AVAILABLE = 30;
+    char buff[MORE_THAN_AVAILABLE];
+    int time_allotted = split2half_rmng_tls_test_time(); // [s]
+    Timer tc_exec_time;
+    tc_exec_time.start();
+
+    TLSSocket sock;
+    TEST_ASSERT_EQUAL(NSAPI_ERROR_OK, sock.set_root_ca_cert(tls_global::cert));
+    if (_tlssocket_connect_to_daytime_srv(sock) != NSAPI_ERROR_OK) {
+        TEST_FAIL();
+        return;
+    }
+    sock.sigio(callback(_sigio_handler, ThisThread::get_id()));
```

Comment:
Not sure, but can we actually construct `Callback` object with two parameters?

Like:
`sock.sigio(callback(osSignalSet, ThisThread::get_id(), SIGNAL_SIGIO);`

So we would not need the `_sigio_handler()` wrapper.

Reply:
There is no such template :(